src/components/signIn/EmailSigninForm.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  ERROR_MESSAGE_USER_ALREADY_EXISTS,
+  ERROR_MESSAGE_USER_DOES_NOT_EXIST,
+} from '@/constants/errors/authentication'
+import {
+  captureClickSubmitEmail,
+  signinTrackEvent,
+} from '@/constants/tracking/pages/signin'
+import Alert from '@/design-system/alerts/alert/Alert'
+import Button from '@/design-system/buttons/Button'
+import TextInput from '@/design-system/inputs/TextInput'
+import { useClientTranslation } from '@/hooks/useClientTranslation'
+import { useCreateVerificationCode } from '@/hooks/verification-codes/useCreateVerificationCode'
+import { useUser } from '@/publicodes-state'
+import type { AuthenticationMode } from '@/types/authentication'
+import { trackEvent, trackPosthogEvent } from '@/utils/analytics/trackEvent'
+import { formatEmail } from '@/utils/format/formatEmail'
+import { isEmailValid } from '@/utils/isEmailValid'
+import { AxiosError } from 'axios'
+import { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+
+type Props = {
+  buttonLabel?: string
+  mode?: AuthenticationMode
+  emailDefaultValue?: string
+}
+
+type FormData = {
+  email: string
+}
+
+export default function EmailSigninForm({
+  buttonLabel,
+  mode,
+  emailDefaultValue,
+}: Props) {
+  const { t } = useClientTranslation()
+
+  const {
+    updateVerificationCodeExpirationDate,
+    updateUserOrganisation,
+    user,
+    updateEmail,
+  } = useUser()
+
+  const {
+    mutateAsync: createVerificationCode,
+    error: errorCreateVerificationCode,
+    isError: isErrorCreateVerificationCode,
+  } = useCreateVerificationCode()
+
+  const errorCode =
+    errorCreateVerificationCode instanceof AxiosError &&
+    errorCreateVerificationCode.response?.data
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    setValue,
+    getValues,
+  } = useForm<FormData>()
+
+  async function onSubmit(data: FormData) {
+    try {
+      const email = formatEmail(data.email)
+
+      // Track the email signin form submission
+      trackEvent(signinTrackEvent(mode))
+      trackPosthogEvent(captureClickSubmitEmail({ mode }))
+
+      const { expirationDate } = await createVerificationCode({
+        email,
+        userId: user.userId,
+        mode,
+      })
+
+      // Reset the organisation local state
+      updateUserOrganisation({
+        administratorEmail: email,
+        slug: '',
+        name: '',
+      })
+
+      // We update the expiration date of the code
+      updateVerificationCodeExpirationDate(expirationDate)
+      updateUserOrganisation({ administratorEmail: email })
+
+      if (!user.email) {
+        updateEmail(email)
+      }
+    } catch (error) {
+      console.log(error)
+      // Error is handled by the useCreateVerificationCode hook
+      return
+    }
+  }
+
+  useEffect(() => {
+    if (emailDefaultValue && getValues('email') === '') {
+      setValue('email', emailDefaultValue)
+    }
+  }, [emailDefaultValue, setValue, getValues, user.email])
```

Comment:
Ce pattern `useEffect` pour mettre à jour une valeur par défaut dérivée des props n'est pas une bonne pratique.

https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

Il n'y a pas une possibilité d'affecter des valeur par défaut dans le useForm par exemple ? Ou dans le register ?

Reply:
Ouais pas fan de devoir faire ça, je comprends pas pourquoi je suis obligé de le faire on peut peut-être regarder ensemble